onadata/apps/api/tests/viewsets/test_xform_submission_viewset.py

```diff
@@ -181,7 +181,7 @@ def test_post_submission_uuid_other_user_username_not_provided(self):
                 auth = DigestAuth('alice', 'bobbob')
                 request.META.update(auth(request.META, response))
                 response = self.view(request)
-                self.assertEqual(response.status_code, 403)
+                self.assertEqual(response.status_code, 404)
```

Comment:
I feel like a `404` here is more appropriate... ? The user used the `/submission` endpoint which in this case means we have to find a form matching the submission data(uuid & id_string) that the user has permission to submit to.

_In this scenario we've located the XForm using `uuid` but the user doesn't have permission to it (Assumption made: `uuid` may not be reliable to find the actual form the user want's to submit to), as such we try to locating the XForm using the id_string within the users owned forms._